EngineLayer/PeptideSpectralMatch.cs

```diff
@@ -269,21 +269,20 @@ public void AddProteinMatch((int, PeptideWithSetModifications) peptideWithNotch)
         /// </summary>
         private static (string ResolvedString, ChemicalFormula ResolvedValue) Resolve(IEnumerable<IEnumerable<Modification>> enumerable)
```

Comment:
There was some weird indentations I took out, so the diff looks a little rough here. I used a temporary list to speed this up in the nested foreach and for getting the first element.